code/API_definitions/quality-on-demand.yaml

```
@@ -950,6 +963,94 @@ components:
             code: NOT_FOUND
             message: "Session Id does not exist"
 
+    Generic405:
+      description: Method Not Allowed
+      headers:
+        x-correlator:
+          $ref: '#/components/headers/x-correlator'
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+          examples:
+            GENERIC_405_METHOD_NOT_ALLOWED:
+              description: Invalid HTTP verb used with a given endpoint
+              value:
+                status: 405
+                code: METHOD_NOT_ALLOWED
+                message: The requested method is not allowed/supported on the target resource.
+
+    Generic406:
+      description: Not Acceptable
+      headers:
+        x-correlator:
+          $ref: "#/components/headers/x-correlator"
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+          examples:
+            GENERIC_406_NOT_ACCEPTABLE:
+              description: API Server does not accept the media type (`Accept-*` header) indicated by API client
+              value:
+                status: 406
+                code: NOT_ACCEPTABLE
+                message: The server cannot produce a response matching the content requested by the client through `Accept-*` headers.
+
+    SessionInConflict409:
+      description: Conflict
+      headers:
+        x-correlator:
+          $ref: '#/components/headers/x-correlator'
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ErrorInfo"
+          example:
+            status: 409
+            code: CONFLICT
+            message: Creating multiple sessions for a device are not allowed.
```

I thought that the API allows for multiple sessions per device.  Would this just be implemented optionally if an operator chooses not to support multiple sessions per device?

What about the case where multiple session are supported for a single device, but the API caller is trying to create a second session for the same flow?  I could see this always generating a conflict error.  In this case, should we support two options `oneOf` for the 409 error code or just support the latter case?

---

@maxl2287 @RandyLevensalor What about the following instead, covering both cases:

```suggestion
    SessionInConflict409:
      description: Conflict
      headers:
        x-correlator:
          $ref: '#/components/headers/x-correlator'
      content:
        application/json:
          schema:
            $ref: "#/components/schemas/ErrorInfo"
          example:
            status: 409
            code: CONFLICT
            message: Conflict with an existing session for the same device.
```

---

@hdamker That's a great way to cover both cases.